abstract/encoding.go

```
@@ -170,13 +193,31 @@ func (de *decoder) value(v reflect.Value, depth int) error {
 			}
 		}
 
-	case reflect.Slice, reflect.Array:
+	case reflect.Slice:
+		if v.IsNil() {
+			panic("slices must be initialized to correct length before decoding")
+		}
+		fallthrough
+	case reflect.Array:
 		l := v.Len()
 		for i := 0; i < l; i++ {
 			if err = de.value(v.Index(i), depth+1); err != nil {
 				return err
 			}
 		}
+
+	case reflect.Int:
```

Yes, I liked my third argument: log.Fail if you encounter an int.
Fail early, fail hard.
But now I know I'd better put int64 on these milliseconds anyway. Or on the clicks for Youtube-videos...